### PR TITLE
Various fixes for the market + Temporal Pagerank implementation

### LINKED ISSUE #3022: Add scipy as dependency for the market
Apparently, scipy is needed as dependency for the market so the docs should be updated.

### LINKED ISSUE #3034: NetworkXError - Linux (devel branch)
- [x] I have read CONTRIBUTING.rst

##### Tribler version/branch+revision: #####
cddfbad065c9a1425e3889f2d45391d15da8480b
##### Operating system and version: #####
Linux `4.11.7-1-ARCH`
##### Steps to reproduce the behavior: #####
Left two official Ubuntu desktop torrents running, for maybe a few minutes, while testing in issue https://github.com/Tribler/tribler/issues/3029
##### Expected behavior: #####
That Tribler runs normally and downloads the torrents.
##### Actual behavior: #####
After sitting at
> 0% Downloading

for a while, it crashed.
##### Relevant log file output: #####
```
Traceback (most recent call last):
  File "/usr/share/tribler/TriblerGUI/event_request_manager.py", line 123, in on_read_data
    raise RuntimeError(json_dict["event"]["text"])
RuntimeError: [Failure instance: Traceback: <class 'networkx.exception.NetworkXError'>: pagerank_scipy: power iteration failed to converge in 100 iterations.
/usr/lib/python2.7/site-packages/twisted/internet/base.py:1243:run
/usr/lib/python2.7/site-packages/twisted/internet/base.py:1252:mainLoop
/usr/lib/python2.7/site-packages/twisted/internet/base.py:878:runUntilCurrent
/usr/lib/python2.7/site-packages/twisted/internet/task.py:239:__call__
--- <exception caught here> ---
/usr/lib/python2.7/site-packages/twisted/internet/defer.py:150:maybeDeferred
/usr/share/tribler/Tribler/community/market/community.py:1251:compute_reputation
/usr/share/tribler/Tribler/community/market/reputation/pagerank_manager.py:29:compute
/usr/lib/python2.7/site-packages/networkx/algorithms/link_analysis/pagerank_alg.py:472:pagerank_scipy
]
```

### LINKED ISSUE #3091: 'NoneType' object has no attribute 'persistence'
##### Tribler version/branch+revision: #####

devel 1be8574c7758395b0a3101100dd58e8e709090c6

##### Operating system and version: #####

Linux Debian mostly stretch

##### Steps to reproduce the behavior: #####

1. Start Tribler core after some hiarus
2. Start Tribler GUI

##### Expected behavior: #####

No errors pops up

##### Actual behavior: #####

Error message is shown, GUI gets closed.

##### Relevant log file output: #####

```
2017-09-13T23:47:25+0300 [twisted.internet.defer#critical] Unhandled error in Deferred:
2017-09-13T23:47:25+0300 [twisted.internet.defer#critical] 
	Traceback (most recent call last):
	  File "/usr/lib/python2.7/dist-packages/twisted/internet/base.py", line 1199, in run
	    self.mainLoop()
	  File "/usr/lib/python2.7/dist-packages/twisted/internet/base.py", line 1208, in mainLoop
	    self.runUntilCurrent()
	  File "/usr/lib/python2.7/dist-packages/twisted/internet/base.py", line 828, in runUntilCurrent
	    call.func(*call.args, **call.kw)
	  File "/usr/lib/python2.7/dist-packages/twisted/internet/task.py", line 239, in __call__
	    d = defer.maybeDeferred(self.f, *self.a, **self.kw)
	--- <exception caught here> ---
	  File "/usr/lib/python2.7/dist-packages/twisted/internet/defer.py", line 150, in maybeDeferred
	    result = f(*args, **kw)
	  File "/home/tribler/tribler/Tribler/community/market/community.py", line 1715, in compute_reputation
	    rep_manager = PagerankReputationManager(self.tradechain_community.persistence.get_all_blocks())
	exceptions.AttributeError: 'NoneType' object has no attribute 'persistence'
	
```

### PR DESCRIPTION
**Pull request by devos50**
This PR contains various fixes for the market. In addition, it contains the implementation of Temporal Pagerank.

Fixes #3034, fixes #3091, fixes #3022

### CODE REVIEW

**Comment by devos50** [comment]
@qstokkink this PR is ready for review. The code that isn't covered by the tests is very hard to reach unfortunately. But correct functionality has been verified with the taxi experiment a while ago so I know it works 👍

File: Tribler/community/market/reputation/temporal_pagerank_manager.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        number_of_nodes = len(personal_nodes)
+        if number_of_nodes == 0:
+            return {}
+        personalisation = {node_name: 1.0 / number_of_nodes if node_name in personal_nodes else 0
+                           for node_name in G.nodes()}
+
+        try:
+            result = nx.pagerank_scipy(G, personalization=personalisation, weight='contribution')
+        except nx.NetworkXException:
+            print("Empty Temporal PageRank, returning empty scores")
```
Comment (qstokkink) [inline Tribler/community/market/reputation/temporal_pagerank_manager.py]: Please use `logging` here.

Comment (devos50) [reply Tribler/community/market/reputation/temporal_pagerank_manager.py]: Oh whoops, forgot that one, let me fix it

**Comment by devos50** [comment]
@qstokkink ready again

**Pull request merged**